addons/network/XEH_preInit.sqf

```diff
@@ -0,0 +1,42 @@
+#include "script_component.hpp"
+
+ADDON = false;
+
+// Restore loadouts lost by the naked unit bug
+[QGVAR(restoreLoadout), {
+    params ["_unit", "_loadout"];
+    if (count _loadout < 7) exitWith {};
+
+    // Check if loadout got lost
+    private _loadoutLost = False;
+    {
+        (_loadout select _forEachIndex) params [["_backup", "", [""]]];
+        // Reject backup if it contains less information
+        if (_backup isEqualTo "" && !(_x isEqualTo "")) exitWith {};
+        if !(_x isEqualTo _backup) exitWith {
+            _loadoutLost = True;
+        };
+    } forEach [
+        primaryWeapon _unit,
+        secondaryWeapon _unit,
+        handgunWeapon _unit,
+        uniform _unit,
+        vest _unit,
+        backpack _unit,
+        headgear _unit
+    ];
+
+    if (_loadoutLost) then {
+        _unit setUnitLoadout _loadout;
+    };
+}] call CBA_fnc_addEventHandler;
+
+["CAManBase", "Local", {
```

Comment:
when someone jips this is triggered for every unit in the game
sending a flood of 150+ events that each include a full loadout array is not good for network traffic

maybe just wait 10 seconds and then add the local EH?

Reply:
This should be improved yeah.

Reply:
Hmm, figuring out a good timing not to mess up jips and still supporting HC transfer scripts is going to be tricky.
Alternatively, we only trigger it when `_local` is `true`. The problem is to figure out on, which machine the units were previously though.

Reply:
It's even worse. Every time a unit gets created, the other machines trigger local as well.
9ca6b6a fixes this.

Reply:
I still don't get this part at all. Shouldn't it literally just be:
```sqf
[QGVAR(restoreLoadout), {
    params ["_unit", "_loadout"];

    if !(getUnitLoadout _unit isEqualTo _loadoutLost) then {
        _unit setUnitLoadout _loadout;
    };
}] call CBA_fnc_addEventHandler;
```
???

Reply:
The name of the event is `restoreLoadout`, but we don't know if it actually restores a loadout. What it actually does is check the loadout and repair it if broken. "validateLoadout"?

Reply:
> The name of the event is `restoreLoadout`, but we don't know if it actually restores a loadout. What it actually does is check the loadout and repair it if broken. "validateLoadout"?

Agreed, `validateLoadout` makes more sense.

> I still don't get this part at all. Shouldn't it literally just be:
> 
> ```sqf
> [QGVAR(restoreLoadout), {
>     params ["_unit", "_loadout"];
> 
>     if !(getUnitLoadout _unit isEqualTo _loadout) then {
>         _unit setUnitLoadout _loadout;
>     };
> }] call CBA_fnc_addEventHandler;
> ```
> 
> ???

Yeah, this will already to. I initially though that we should just check a subset of items like it was done for the ACEX implementation. Moreover, the "Reject backup if it contains less information" part is actually no longer needed, since I noticed that this exception no longer occurs, since the rogue trigger was dealt with.